tests/test_metadata.py

```diff
@@ -74,7 +86,74 @@ def test_preauthencoding(self):
         envelope_obj = Envelope.from_dict(copy.deepcopy(self.envelope_dict))
 
         # Checking for Pre-Auth-Encoding generated is correct.
-        self.assertEqual(self.pae, envelope_obj.pae)
+        self.assertEqual(self.pae, envelope_obj.pae())
+
+    def test_sign(self):
+        """Test payload signing of envelope."""
+
+        # Create an Envelope with no signatures.
+        envelope_dict = copy.deepcopy(self.envelope_dict)
+        envelope_dict["signatures"] = []
+        envelope_obj = Envelope.from_dict(envelope_dict)
+
+        for (_, key_dict) in self.key_dicts:
+            # Test for invalid scheme.
+            valid_scheme = key_dict["scheme"]
+            key_dict["scheme"] = "invalid_scheme"
+            signer = SSlibSigner(key_dict)
+            with self.assertRaises((FormatError, UnsupportedAlgorithmError)):
+                envelope_obj.sign(signer)
+
+            # Sign the payload.
+            key_dict["scheme"] = valid_scheme
+            signer = SSlibSigner(key_dict)
+            envelope_obj.sign(signer)
+
+        # Tests for signatures created.
+        self.assertEqual(len(self.key_dicts), len(envelope_obj.signatures))
+        for signature in envelope_obj.signatures:
+            self.assertIsInstance(signature, Signature)
```

Comment:
It seems like everything you are testing here is also tested below in `test_verify`, except for the `"invalid_scheme"` test, which IMO is a better fit for `test_key` (#4). 
What do you think about removing `test_sign` and calling the method below `test_sign_and_verify`?